src/main/java/net/dv8tion/jda/api/requests/restaction/InviteAction.java

```diff
@@ -118,4 +120,100 @@
     @Nonnull
     @CheckReturnValue
     InviteAction setUnique(@Nullable final Boolean unique);
+
+    /**
+     * Sets the type of target for this voice channel invite.
+     *
+     * @param type
+     *        The target type for this voice channel invite.
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If this channel is not a voice channel.
+     *
+     * @return The current InviteAction for chaining.
+     */
+    @Nonnull
+    @CheckReturnValue
+    InviteAction setTargetType(@Nullable final Invite.TargetType type);
+
+    /**
+     * Sets the id of the targeted application.
+     * This will automatically {@link #setTargetType(Invite.TargetType) set the target type} to {@link Invite.TargetType#EMBEDDED_APPLICATION}.
+     *
+     * @param applicationId
+     *        The id of the embedded application or {@code null} or {@link Invite.TargetType#UNKNOWN} to use none.
+     *        Requires type {@link Invite.TargetType#EMBEDDED_APPLICATION}
+     *
+     * @return The current InviteAction for chaining.
+     */
+    @Nonnull
+    @CheckReturnValue
+    InviteAction setTargetApplication(@Nullable final Long applicationId);
```

Comment:
Making two methods with the same name accept nullable parameters is bad since you run into ambiguity. I'd just use `0` instead of null for this.